### PR TITLE
Add support for Workers APIs

### PR DESCRIPTION
**Pull request by hankjacobs**
This PR adds support for the Workers suite of APIs (https://developers.cloudflare.com/workers/api/ and https://developers.cloudflare.com/workers/api/config-api-for-enterprise/).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 node_modules/
 coverage/
 .nyc_output/
+.vscode/
```
Comment (terinjokes) [inline .gitignore]: This is not related to this project, and can be ignored locally.

File: .travis.yml [hunk 2]
```diff
@@ -14,19 +12,29 @@ test_stage: &test_stage
 jobs:
   include:
     - stage: lint
+      before_install:
+        - npm install -g npm@latest
       node_js: 8
       script:
         - npm run lint
     - stage: test
+      before_install:
```
Comment (terinjokes) [inline .travis.yml]: IIRC, `before_install` can be included in the reference, then overridden for this specific version.

File: lib/method.js [hunk 3]
```diff
@@ -75,6 +75,8 @@ const identity = function identity(x) {
 module.exports = function(spec) {
   const requestMethod = (spec.method || 'GET').toUpperCase();
   const encode = spec.encode || identity;
+  const json = spec.json !== false;
+  const contentType = spec.contentType;
```
Comment (terinjokes) [inline lib/method.js]: Do you mind also setting the default value here (and updating the values in the test), like you have have for `json`?

File: lib/resources/EnterpriseZoneWorkersRoutes.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+     *
+     * @function browse
+     * @memberof EnterpriseZoneWorkersRoutes
+     * @instance
+     * @async
+     * @param {string} zone_id - The zone ID
+     * @returns {Promise<Object>} The route browse response object.
+     */
+    /**
+     * read allows for retrieving a specific zone's workers route
```
Comment (terinjokes) [inline lib/resources/EnterpriseZoneWorkersRoutes.js]: The documentation doesn't say you can get a specific route, so we might need to check this. @rita3ko

File: lib/resources/ZoneWorkers.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+     *
+     * @function validate
+     * @memberof ZoneWorkers
+     * @instance
+     * @async
+     * @param {string} zoneId - The zone ID
+     * @param {string} script - The worker script
+     * @returns {Promise<Object>} The validate response object.
+     */
+    validate: method({
```
Comment (terinjokes) [inline lib/resources/ZoneWorkers.js]: I'm hesitant including endpoints not documented. @rita3ko, mind taking a look and seeing if we can't document this?

File: lib/resources/ZoneWorkersRoutes.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+     *
+     * @function browse
+     * @memberof ZoneWorkersRoutes
+     * @instance
+     * @async
+     * @param {string} zone_id - The zone ID
+     * @returns {Promise<Object>} The route browse response object.
+     */
+    /**
+     * read allows for retrieving a specific zone's workers route
```
Comment (terinjokes) [inline lib/resources/ZoneWorkersRoutes.js]: Same thing we'll need to double check here. @rita3ko

File: test/method.js [hunk 7]
```diff
@@ -240,4 +250,39 @@ describe('method', () => {
       }
     ).then(resp => assert.deepEqual(resp, body));
   });
+
+  it('should set json when specified', () => {
```
Comment (terinjokes) [inline test/method.js]: Do you mind adding a case for setting `content-type`?

File: index.js [hunk 8]
```diff
@@ -20,6 +20,11 @@ const resources = {
   zones: require('./lib/resources/Zones'),
   zoneSettings: require('./lib/resources/ZoneSettings'),
   zoneCustomHostNames: require('./lib/resources/ZoneCustomHostNames'),
+  zoneWorkers: require('./lib/resources/ZoneWorkers'),
+  zoneWorkersScript: require('./lib/resources/ZoneWorkersScript'),
+  zoneWorkersRoutes: require('./lib/resources/ZoneWorkersRoutes'),
+  enterpriseZoneWorkersScripts: require('./lib/resources/EnterpriseZoneWorkersScripts'),
```
Comment (terinjokes) [inline index.js]: This is alphabetical.

**Comment by hankjacobs** [comment]
Feedback implemented. Thanks for the prompt review! I'll wait for @rita3ko's feedback on the undocumented APIs.

File: lib/resources/ZoneWorkers.js (hunk 5 above)
Comment (rita3ko) [reply lib/resources/ZoneWorkers.js]: We just need to add to docs. OK with this.

File: lib/resources/EnterpriseZoneWorkersRoutes.js (hunk 4 above)
Comment (rita3ko) [reply lib/resources/EnterpriseZoneWorkersRoutes.js]: Same. Just need to add to docs. Will do so shortly.

**Comment by terinjokes** [comment]
Thanks @rita3ko. I'll do another review now.

**Comment by terinjokes** [comment]
Splitting and merging the patch locally.

**Comment by terinjokes** [comment]
Merged as 92af8b025a795c815154a380f27bd920c881cb1b.
